medschool/src/parse.rs

```diff
@@ -279,48 +292,59 @@ fn dfs_fields<'a, const MAX_RECURSION_LEVEL: usize>(
 /// be where we resolve all references and return the same HashSet and let the caller
 /// decide what the root should be.
 #[tracing::instrument(level = "trace", ret)]
-pub fn resolve_references(types: HashSet<PartialType>) -> Option<PartialType> {
-    /// Maximum recursion level for safety.
+pub fn resolve_references(
+    types: HashSet<PartialType>,
+    schema: Option<JSONSchema>,
+) -> Result<Option<PartialType>, DocsError> {
+    // Maximum recursion level for safety.
     const MAX_RECURSION_LEVEL: usize = 10;
     // Cache to perform memoization between recursive calls so we don't have to resolve the same
     // type multiple times. Mapping between `ident` -> `resolved_docs`.
     // For example, if we have a types [`A`, `B`, `C`] and A has a field of type `B` and `B` has a
     // field of type `C`, and `C` has already been resolved, we don't want to resolve `C` again
     // as we iterate over the types. A -> (B -> C), (B -> C), (C)
     let mut cache = HashMap::with_capacity(types.len());
+    let mut max_area = 0;
```

Comment:
Same thing here, the logic looks pretty much the same, except it's more `Result`-y now? You could use `try_collect` for example.